From 1dcb4461efdd89f7b41b4e2303ce03795c6a7de4 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 10 Jul 2024 16:07:56 +0200
Subject: [PATCH 1037/1695] plat-stm32mp2: set CFG_RESERVED_VASPACE_SIZE to 32
 MB for remoteproc support

The remoteproc framework maps the memory regions declared in the device
tree to be able to write the remote processor code but also to clean the
memory when stopping the remote processor.
Increase the CFG_RESERVED_VASPACE_SIZE that defines the size of the virtual
address space to support the mapping of an 8MB memory region in DDR.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 353f594ff0e6db87fb15949728a2d45c93520512 plat-stm32mp2: set CFG_RESERVED_VASPACE_SIZE to 32 MB for remoteproc support]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I1f13bd084896507da7c454c6e7ad22dab0de4004
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/392630
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index f25b2be04..a2f6f7318 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -218,6 +218,9 @@ ifeq ($(CFG_REMOTEPROC_PTA),y)
 CFG_IN_TREE_EARLY_TAS += remoteproc/80a4c275-0a47-4905-8285-1486a9771a08
 # Embed public part of this key in OP-TEE OS
 RPROC_SIGN_KEY ?= keys/default.pem
+# Increase the RESERVED VA SPACE to be able to map reserved-memory regions
+# assigned to the remote processor.
+CFG_RESERVED_VASPACE_SIZE = 32 * 1024 * 1024
 endif
 
 # Default enable HWRNG PTA support
-- 
2.47.2

